local-cli/cliEntry.js

```diff
@@ -8,154 +8,82 @@
  */
 'use strict';
 
-const bundle = require('./bundle/bundle');
-const childProcess = require('child_process');
+const cli = require('commander');
+
 const Config = require('./util/Config');
-const defaultConfig = require('./default.config');
-const dependencies = require('./dependencies/dependencies');
-const generate = require('./generate/generate');
-const library = require('./library/library');
-const path = require('path');
-const Promise = require('promise');
-const runAndroid = require('./runAndroid/runAndroid');
-const runIOS = require('./runIOS/runIOS');
-const server = require('./server/server');
-const TerminalAdapter = require('yeoman-environment/lib/adapter.js');
-const yeoman = require('yeoman-environment');
-const unbundle = require('./bundle/unbundle');
-const upgrade = require('./upgrade/upgrade');
-const version = require('./version/version');
+const childProcess = require('child_process');
 
+const path = require('path');
 const fs = require('fs');
 const gracefulFs = require('graceful-fs');
 
+const init = require('./init/init');
+const commands = require('./commands');
+const assertRequiredOptions = require('./util/assertRequiredOptions');
+const pkg = require('../package.json');
+const defaultConfig = require('./default.config');
+
 // graceful-fs helps on getting an error when we run out of file
 // descriptors. When that happens it will enqueue the operation and retry it.
 gracefulFs.gracefulify(fs);
 
-const documentedCommands = {
-  'start': [server, 'starts the webserver'],
-  'bundle': [bundle, 'builds the javascript bundle for offline use'],
-  'unbundle': [unbundle, 'builds javascript as "unbundle" for offline use'],
-  'new-library': [library, 'generates a native library bridge'],
-  'android': [generateWrapper, 'generates an Android project for your app'],
-  'run-android': [runAndroid, 'builds your app and starts it on a connected Android emulator or device'],
-  'run-ios': [runIOS, 'builds your app and starts it on iOS simulator'],
-  'upgrade': [upgrade, 'upgrade your app\'s template files to the latest version; run this after ' +
-                       'updating the react-native version in your package.json and running npm install']
-};
+cli.version(pkg.version);
 
-const exportedCommands = {dependencies: dependencies};
-Object.keys(documentedCommands).forEach(function(command) {
-  exportedCommands[command] = documentedCommands[command][0];
-});
+const defaultOptParser = (val) => val;
 
-const undocumentedCommands = {
-  '--version': [version, ''],
-  'init': [printInitWarning, ''],
+const handleError = (err) => {
+  console.error();
+  console.error(err.message || err);
```

Comment:
Ideally I'd like to reject promises with an error, not a plain string. Just a backwards compatible thing until we move it all.